src/inmanta/data/__init__.py

```diff
@@ -1296,7 +1296,12 @@ def get_connection(
             return util.nullcontext(connection)
         # Make pypi happy
         assert cls._connection_pool is not None
-        return cls._connection_pool.acquire()
+        try:
+            return cls._connection_pool.acquire(timeout=1.0)
+        except asyncio.TimeoutError:
+            # record exhausted pool event
+            util.ExhaustedPoolWatcher.record_event()
+            return cls._connection_pool.acquire()
```

Comment:
> But I don't really know how asyncpg behaves in that situation. Did you investigate this?

I'm not following 100% what do you mean ?

Reply:
>  It would be nice if we can also detect the situation where the pool size on the database server is configured to be lower than the pool size on the inmanta server

I added a check in the `check_for_pool_exhaustion` method

Reply:
PostgreSQL has a config option to indicate the maximum amount of connections it accepts (See: max_connections config option [here](https://www.postgresql.org/docs/13/runtime-config-connection.html)). So it's possible that max_connection < connection-pool-max-size. But this is something that should be handled by the monitoring/alerting of the PostgreSQL server, so you can ignore that comment.